File: swift/Sources/ShapesBaseModels/Rectangle.swift
import Foundation
import ShapesBaseCommons

public final class Rectangle: Dimensional {
    private let width: Double
    private let height: Double

    public init(_ width: Double, _ height: Double) {
        let dimOne: Double = width > 0 ? width : 1
        let dimTwo: Double = height > 0 ? height : 1
        self.width = dimOne
        self.height = dimTwo
    }

    public func area() -> Double {
        return width * height
    }

    public func perimeter() -> Double {
        return 2 * (width + height)
    }
}

File: csharp/src-temp/Shapes/Base/Models/Rectangle.cs
using Shapes.Base.Commons;

namespace Shapes.Base.Models;

public sealed class Rectangle : IDimensional {
    private readonly double _width;
    private readonly double _height;

    public Rectangle(double width, double height) {
        double dimOne = width > 0 ? width : 1;
        double dimTwo = height > 0 ? height : 1;
        _width = dimOne;
        _height = dimTwo;
    }

    public double Area() {
        return _width * _height;
    }

    public double Perimeter() {
        return 2 * (_width + _height);
    }
}
